guides/keras_nlp/semantic-similarity-with-bert.py

```diff
@@ -0,0 +1,216 @@
+print(valid_df.similarity.value_counts())
+
+# Filter and shuffle train and validation data
+train_df = (
+    train_df[train_df.similarity != "-"]
+    .sample(frac=1.0, random_state=42)
+    .reset_index(drop=True)
+)
+valid_df = (
+    valid_df[valid_df.similarity != "-"]
+    .sample(frac=1.0, random_state=42)
+    .reset_index(drop=True)
+)
+
+# Convert label column to numerical values for train, validation and test data
+train_df["label"] = train_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_train = train_df.label
+
+valid_df["label"] = valid_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_val = valid_df.label
+
+test_df["label"] = test_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_test = test_df.label
+
+"""
+Initialize a BertPreprocessor object from KerasNLP library with preset 
+configuration "bert_tiny_en_uncased". The preprocessor can preprocess 
+sentence pairs for BERT-based models by converting tokens to corresponding IDs in 
+BERT vocabulary and adding special tokens.
+"""
+preprocessor = BertPreprocessor.from_preset("bert_tiny_en_uncased")
+
+"""
+Preprocessing the sentence pairs in the training, validation, and test data using 
+a preprocessor object. BertPreprocessor automatically takes care of sentence packing,
+and seperates them with [SEP] token
+"""
+x_train = preprocessor((train_df["sentence1"], train_df["sentence2"]))
+x_val = preprocessor((valid_df["sentence1"], valid_df["sentence2"]))
+x_test = preprocessor((test_df["sentence1"], test_df["sentence2"]))
+
+"""
+### Train the Model End to End
+ 
+We'll use BertClassifier from KerasNLP. This model attaches a classification head to 
+a keras_nlp.model.BertBackbone backbone, mapping from the backbone outputs to logit 
+output suitable for a classification task. This makes the job a lot easier !
+
+Here we'll use this model with pre-trained weights. `from_preset()` method allows you
+to use your own preprocessor. Here we'll set the `num_classes` as 3 for SNLI dataset
+"""
+
+model = BertClassifier.from_preset(
+    "bert_tiny_en_uncased", num_classes=3, preprocessor=None
+)
+
+model.compile(
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    optimizer=tf.keras.optimizers.Adam(1e-5),
+    metrics=["accuracy"],
+)
+
+model.summary()
+
+"""
+Training the model we just compiled using the fit() method with the following arguments:
+
+    x_train: The input data for the training set, preprocessed using BertPreprocessor
+    y_train: The target labels for the training set
+    validation_data: A tuple containing the input and target data for the validation set
+    epochs: The number of epochs to train the model
+    batch_size: The number of samples per gradient update
+"""
+model.fit(x=x_train, y=y_train, validation_data=(x_val, y_val), epochs=4, batch_size=64)
```

Comment:
I wonder if it's worth a plug to show how easy it would be to switch to other models to boost performance. E.g. show setting up a run with `roberta_base_en` and talk about how much that would improve things.